include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -0,0 +1,242 @@
+/* Copyright 2021 Felice Pantaleo, Andrea Bocci
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/atomic/Traits.hpp>
+#include <alpaka/core/BoostPredef.hpp>
+
+#include <boost/atomic.hpp>
+
+#include <array>
+#include <type_traits>
+
+namespace alpaka
+{
+    //! The atomic ops based on atomic_ref for CPU accelerators.
+    //
+    //  Atomics can be used in the grids, blocks and threads hierarchy levels.
+    //
+
+    class AtomicAtomicRef
+    {
+    };
+
+    namespace traits
+    {
+        //! The CPU accelerators AtomicAdd.
+        template<typename T, typename THierarchy>
+        struct AtomicOp<AtomicAdd, AtomicAtomicRef, T, THierarchy>
+        {
+            ALPAKA_FN_HOST static auto atomicOp(AtomicAtomicRef const& atomic, T* const addr, T const& value) -> T
+            {
+                static_assert(
+                    std::is_trivially_copyable_v<T> && boost::atomic_ref<T>::required_alignment <= alignof(T),
+                    "Type not supported by AtomicAtomicRef, please recompile defining "
+                    "ALPAKA_DISABLE_ATOMIC_ATOMICREF.");
```

Comment:
Could you please refactor that check into a separate function to eliminate some of the redundancy in the following functions?
```c++
template <typename T>
void isSupportedByAtomicAtomicRef() {
    static_assert(
        std::is_trivially_copyable_v<T> && boost::atomic_ref<T>::required_alignment <= alignof(T),
        "Type not supported by AtomicAtomicRef, please recompile defining "
        "ALPAKA_DISABLE_ATOMIC_ATOMICREF.");
}
```
```suggestion
                isSupportedByAtomicAtomicRef<T>();
```